server/db/models/users.js

```diff
@@ -0,0 +1,258 @@
+//define Users model here
+const Sequelize = require('sequelize');
+const { db } = require('./../database.js');
+
+const { UUID, UUIDV4, STRING, BOOLEAN, INTEGER, DATEONLY } = Sequelize;
+
+const Users = db.define('users', {
+    userId: {
+        primaryKey: true,
+        type: UUID,
+        defaultValue: UUIDV4,
+    },
+    userType: {
+        type: STRING,
+        allowNull: false,
+        //may need to update this later depends on how a user chooses his type
+        validate: {
+            isIn: [
+                ['Guest', 'Existing customer', 'Admin']
+            ],
+        },
+    },
+    loggedIn: {
+        type: BOOLEAN,
+    },
+    email: {
+        type: STRING,
+        allowNull: false,
+        unique: true,
+        validate: {
+            notEmpty: {
+                args: true,
+                msg: 'Email address cannot be empty',
+            },
+            notNull: {
+                arg: true,
+                msg: 'Email address cannot be null',
+            },
+            isEmail: {
+                arg: true,
+                msg: 'Email address must be valid',
+            }
+        },
+    },
+    password: {
+        //password is not required to guest, but required to admin and existing customer
+        //need to figure out how to make it notNull when userType is "Admin" or "Existing Customer"
+        type: STRING,
+        validate: {
+            len: {
+                arg: 6,
+            }
+        }
+    },
+    firstName: {
+        type: STRING,
+        allowNull: false,
+        validate: {
+            notEmpty: {
+                args: true,
+                msg: 'First name cannot be empty',
+            },
+            notNull: {
+                arg: true,
+                msg: 'First name cannot be null',
+            },
+        },
+    },
+    lastName: {
+        type: STRING,
+        allowNull: false,
+        validate: {
+            notEmpty: {
+                args: true,
+                msg: 'Last name cannot be empty',
+            },
+            notNull: {
+                arg: true,
+                msg: 'Last name cannot be null',
+            },
+        },
+    },
+    phone: {
+        type: INTEGER,
+        validate: {
+            isNumeric: {
+                args: true,
+                msg: 'Phone number should only contain numbers',
+            },
+            len: {
+                arg: 10,
+            }
+        }
+    },
+    shippingAddress: {
```

Comment:
all shipping credit card and billing can allow null. Users won't need to save an address or credit card on file until their first checkout.